From 5fc0429e6867633acf3f4aff3120f0aa8b71d796 Mon Sep 17 00:00:00 2001
From: Doug Nazar <nazard@nazar.ca>
Date: Sun, 23 Jul 2017 06:20:47 -0400
Subject: [PATCH] MXC-CEC: Return writable for poll when no link status.

If there is no link we need to allow writes otherwise libcec will
block on select() in the SocketWrite() call. The write() will still
fail with EAGAIN.

Signed-off-by: Doug Nazar <nazard@nazar.ca>
---
 drivers/mxc/hdmi-cec/mxc_hdmi-cec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/hdmi-cec/mxc_hdmi-cec.c b/drivers/mxc/hdmi-cec/mxc_hdmi-cec.c
index 5519ea1..527c8f2 100644
--- a/drivers/mxc/hdmi-cec/mxc_hdmi-cec.c
+++ b/drivers/mxc/hdmi-cec/mxc_hdmi-cec.c
@@ -584,7 +584,7 @@ static unsigned int hdmi_cec_poll(struct file *file, poll_table *wait)
 	poll_wait(file, &rx_queue, wait);
 	poll_wait(file, &tx_queue, wait);
 
-	if (priv->link_status == 1 &&
+	if (priv->link_status == 0 ||
 	    priv->tx_answer == CEC_TX_AVAIL)
 		mask |= POLLOUT | POLLWRNORM;
 
-- 
2.7.4

